src/net/tcp/tcp.cpp

```diff
@@ -131,7 +131,7 @@ seq_t TCP::generate_iss() {
 */
 port_t TCP::next_free_port() {
 
-  current_ephemeral_ = (current_ephemeral_ == 0) ? current_ephemeral_ + 1025 : current_ephemeral_ + 1;
+  current_ephemeral_ = (current_ephemeral_ == port_ranges::DYNAMIC_END) ? port_ranges::DYNAMIC_START : current_ephemeral_ + 1;
 
   // Avoid giving a port that is bound to a service.
   while(listeners_.find(current_ephemeral_) != listeners_.end())
```

Comment:
I think you're right here yes. But it shouldn't be a listener on one of the dynamic ports (we do not prevent this tho, and I'm not sure if we want to do this either?). I know the TCP port-handling is not bulletproof, and that it maybe should be reworked some. 
What makes it a bit troublesome is that there are two collections where a port could be taken; `listeners_` (listening bound ports) and `connections_` (outgoing connections).
Another bug is when all dynamic ports are taken/active, we start overwriting existing ones. The port-handling in UDP is much more simpler/safer.